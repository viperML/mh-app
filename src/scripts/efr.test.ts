import { expect, test } from "vitest";
import { efr } from "./efr";
import { mergeSkillRanks, type SkillRank2 } from "./skill";

function fakeSkill(name: string, level: number): SkillRank2 {
    return {
        level,
        skill: {
            name,
            id: 0,
            description: "FIXME",
            kind: "armor",
        },
    };
}

test("merge skills", () => {
    expect(
        mergeSkillRanks(fakeSkill("Weakness Exploit", 3), fakeSkill("Agitator", 4), fakeSkill("Agitator", 1)),
    ).toMatchSnapshot();
});

function testSkill(...skills: SkillRank2[]) {
    // Create a test name from all the skills
    const name = skills.map(rank => `${rank.skill.name}@${String(rank.level)}`).join("+");

    test(`Skills: ${name}`, () => {
        expect(
            efr({
                attack: 100,
                affinity: 0,
                skills: mergeSkillRanks(...skills),
            }),
        ).toMatchSnapshot();
    });
}

testSkill();

testSkill(fakeSkill("Agitator", 5));
testSkill(fakeSkill("Weakness Exploit", 5));
testSkill(fakeSkill("Critical Eye", 5));
