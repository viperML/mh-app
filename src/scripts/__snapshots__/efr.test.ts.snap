// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skills:  1`] = `
{
  "affinity": 0,
  "attack": 100,
  "efr": 100,
}
`;

exports[`Skills: Agitator@5 1`] = `
{
  "affinity": 0.15,
  "attack": 120,
  "efr": 124.5,
}
`;

exports[`Skills: Critical Eye@5 1`] = `
{
  "affinity": 0.2,
  "attack": 100,
  "efr": 105,
}
`;

exports[`Skills: Weakness Exploit@5 1`] = `
{
  "affinity": 0.3,
  "attack": 100,
  "efr": 107.5,
}
`;

exports[`merge skills 1`] = `
Map {
  "Weakness Exploit" => 3,
  "Agitator" => 5,
}
`;
